src/sage/matrix/matrix2.pyx

```diff
@@ -18618,12 +18618,25 @@ def _matrix_power_symbolic(A, n):
          -1/2*I*(a + I*b)^k + 1/2*I*(a - I*b)^k,
          1/2*I*(a + I*b)^k - 1/2*I*(a - I*b)^k,
          1/2*(a + I*b)^k + 1/2*(a - I*b)^k]
+
+    Check if :trac:`36838` is fixed:Checking symbolic power of
+    nilpotent matrix::
+
+        sage: A = matrix([[0,1],[0,0]]); A
+        [0 1]
+        [0 0]
+        sage: n = var('n'); n
+        n
+        sage: An = A^n; An
+        [        0^n 0^(n - 1)*n]
+        [          0         0^n]
```

Comment:
According to the comment on line 18678,
We would then have something like
`       [        kronecker_delta(0, n) n*kronecker_delta(1, n)]`
`       [        0                     kronecker_delta(0, n)]`